wireapp/xenon
# feat(federation): [WPB-11008] Change user identifier from UUID to UUID+domain(String)

* Add basic encryption test
* Add javadoc on key classes
* Adapt API and client interfaces to also work with fully-qualified userIds
* Update dependency for security reasons
* Write README.md project description

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Add federation support, change user ids from UUID to UUID+String

### Solutions

Improve Qualified class and use it in the the classes and interfaces

### Testing

#### Test Coverage (Optional)


#### How to Test

Use the updated Xenon in a client (like Helium library or Lithium), adapt them, and check that major operations still work (sending messages, getting users' data...

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
